File: ts/256/206.ts
import { ListNode } from '../class'

// 迭代
export function reverseList(head: ListNode | null): ListNode | null {
  let pre = null, cur = head, next = null
  while (cur) {
    next = cur.next
    cur.next = pre
    pre = cur
    cur = next
  }
  return pre
};

File: c/256/206.c
#include <stdio.h>
#include "../structs.h"


struct ListNode* reverseList(struct ListNode* head){
  struct ListNode* pre = NULL;
  struct ListNode* cur = head;
  struct ListNode* nxt;
  while (cur) {
    nxt = cur->next;
    cur->next = pre;
    pre = cur;
    cur = nxt;
  }
  return pre;
}